### DISCUSSION TITLE
技巧分享：Alfred Workflows 调用 Saladict 实现全局【文本翻译】

### DISCUSSION THREAD

**Discussion by xinshangshangxin** [crimx/ext-saladict#509] [Show and tell]
## 下载

[saladict_v1.1](https://github.com/crimx/ext-saladict/files/3711425/saladict.alfredworkflow.zip)

## 效果图
![Jietu20191010-154129](https://user-images.githubusercontent.com/8779091/66548810-8fa1f380-eb74-11e9-8177-8812a0a385bd.gif)

![Jietu20191010-162724](https://user-images.githubusercontent.com/8779091/66551929-fc1ff100-eb7a-11e9-9785-63693bcffd05.gif)

## 使用/配置

1. 开启独立窗口快捷查词
  ![image](https://user-images.githubusercontent.com/8779091/66549512-025f9e80-eb76-11e9-9a1c-156051b9116a.png)

2. [设置Saladict 独立窗口快捷键](https://github.com/crimx/ext-saladict/wiki#%E6%94%AF%E6%8C%81%E5%BF%AB%E6%8D%B7%E9%94%AE%E6%93%8D%E4%BD%9C) 为 `ctrl` + `cmd` + `L`

  ![image](https://user-images.githubusercontent.com/8779091/66553493-caf4f000-eb7d-11e9-84e8-933d6abf3985.png)

3. 解压上面下载的 `saladict.alfredworkflow.zip`, 双击 `saladict.alfredworkflow` 导入至 `alfred workflow`

4. 双击 `hotkey` 录入快捷键
![image](https://user-images.githubusercontent.com/8779091/66549727-83b73100-eb76-11e9-8059-8bd1dbc4e920.png)

5. 选中文字, 触发快捷键即可 调用沙拉查询

### 以下为可选配置

6.  `Saladict 独立窗口快捷键` 设置的如果不是 `ctrl` + `cmd` + `L`, 则在 `alfred` 中双击 `Run NSAppleScript`
   ![image](https://user-images.githubusercontent.com/8779091/66549946-004a0f80-eb77-11e9-9691-c99e72e2bc69.png)

修改截图所示代码, 其中 的 code 对应表请查看 [complete-list-of-applescript-key-codes](https://eastmanreference.com/complete-list-of-applescript-key-codes)

![image](https://user-images.githubusercontent.com/8779091/66553753-4191ed80-eb7e-11e9-8890-50fafc53a68a.png)

比如 想要修改成 `option` + `T`, 那就把
`key code 37 using {control down, command down}` 改成
`key code 17 using {option down}`

## 说明
感谢作者针对 https://github.com/crimx/ext-saladict/issues/485 进行加强, 才能使本脚本工作

## 参考文档 
- [手把手教你用 AppleScript 模拟鼠标键盘操作，实现 macOS 系统的自动化操作
](https://sspai.com/post/43758)
- [Complete list of AppleScript key codes](https://eastmanreference.com/complete-list-of-applescript-key-codes)

**Comment by crimx** [comment]
谢谢，非常棒的分享！

**Reply by YuanchenHua** [reply]
新增了一个功能，查完单词，最前端不是沙拉单词弹窗，会将弹窗自动关闭功能，避免需要反复粘贴才能查询，mac M3 14.2 (23C64) 系统可用。

新增功能，在查询完单词后，如果最前端窗口不再是沙拉单词弹窗（chrome app）则自动关闭刚刚
TODO：最好能主动查询到哪个窗口是沙拉单词弹窗，指定关闭，目前方法是最新一个chrome window，用起来有一点点问题，感兴趣的小伙伴可以试试看，以及求大佬优化 
[沙拉1.1.2.zip](https://github.com/crimx/ext-saladict/files/15272861/1.1.2.zip)

**Reply by YuanchenHua** [reply]
我自己设置快捷键是 cmd+t， 划词，cmd+t 即可完成查询，然后重新点击文档（比如pdf），沙拉单词的窗口不在最前端，即会自动关闭窗口。 下一次有新的要查询的词，只要  划词，cmd+t 即可

**Reply by YuanchenHua** [reply]
在热键触发以后新增一个自动关闭的AppleScript脚本
<img width="662" alt="截屏2024-05-10 17 11 25" src="https://github.com/crimx/ext-saladict/assets/45629278/389672f8-7ffe-4eb5-9a94-be19011fd4e5">

```
on alfred_script(q)
    delay 5
    repeat
        tell application "System Events"
            -- 检查Chrome是否是最前端应用
            set chromeIsFrontmost to frontmost of application process "Google Chrome"
        end tell
        -- 如果Chrome不是最前端应用
        if chromeIsFrontmost is false then
            tell application "Google Chrome"
                -- 获取窗口的数量
                set numberOfWindows to count windows
                if numberOfWindows > 0 then
                    -- 获取最后一个窗口（通常是最新创建的）
                    set newestWindow to first item of windows
                    -- 关闭这个窗口
                    close newestWindow
                end if
            end tell
            -- 退出脚本
            exit repeat
        end if
        -- 暂时延迟1秒再次检查
        delay 1
    end repeat
end alfred_script
```

**Reply by YuanchenHua** [reply]
更新了一下脚本，采取window title的方式进行检查关闭，避免勿关
[沙拉1.1.3.zip](https://github.com/crimx/ext-saladict/files/15281484/1.1.3.zip) 1.1.3版本快捷键更新为双击 cmd，避免原来原快捷（cmd+t）和一些软件默认快捷键冲突问题
```
on alfred_script(q)
    delay 5
    repeat
        tell application "System Events"
            set chromeIsFrontmost to frontmost of application process "Google Chrome"
        end tell

        -- 如果Chrome不是最前端应用
        if chromeIsFrontmost is false then
            tell application "Google Chrome"
                set foundWindow to false
                -- 遍历所有窗口
                repeat with theWindow in every window
                    if title of theWindow is "沙拉查词-独立查词窗口" then
                        close theWindow
                        set foundWindow to true
                        exit repeat
                    end if

                end repeat
                
                -- 如果没有找到指定的窗口，则退出循环
                if foundWindow is false then
                    exit repeat
                end if
            end tell
        end if
        
        -- 暂时延迟1秒再次检查
        delay 1
    end repeat
end alfred_script
```

**Comment by crimx** [comment]
新增的快捷键可以直接让沙拉查词查询剪贴板，不知对你的 workflow 有没有帮助。

**Comment by xinshangshangxin** [comment]
可以大幅度简化脚本, 回头我更新下

**Comment by xinshangshangxin** [comment]
已更新成1.1 版本, 调用 `新增的快捷键` 来实现

**Comment by nb5p** [comment]
> 新增的快捷键可以直接让沙拉查词查询剪贴板，不知对你的 workflow 有没有帮助。

这个只能查剪切板的不太方便，有没有可能改成接受脚本传来的参数

**Comment by crimx** [comment]
我可以让沙拉查词响应事件，如果你有兴趣欢迎另开 issue 我们讨论一下接口设计。

**Comment by crimx** [comment]
@qq645381995 请不要搭楼反馈不相关问题。

**Comment by crazyair** [comment]
![image](https://user-images.githubusercontent.com/7971419/67857907-e9ce1d00-fb52-11e9-9c96-631f7d86f875.png)
主要我这怎么没反应呢？

**Reply by jianJizz** [reply]
我也遇到这个问题了，我的ctrl+cmd+L可以在浏览器器外调起，但是在alfred就不行,
![image](https://user-images.githubusercontent.com/24221653/114170156-15b71a00-9965-11eb-8535-785358752e0e.png)

**Comment by nb5p** [comment]
> ![image](https://user-images.githubusercontent.com/7971419/67857907-e9ce1d00-fb52-11e9-9c96-631f7d86f875.png)
> 主要我这怎么没反应呢？

建议仔细阅读，需要设置快捷键，使用快捷键呼出沙拉查词并检索剪切板中的内容

**Comment by youcanping** [comment]
5. 选中文字, 触发快捷键即可 调用沙拉查询
这个怎么设置，我再pdf中选中单词没有沙拉图标，需要快捷键 ctrl + cmd + L 才出来

**Reply by muziyuqi** [reply]
同没看明白==||

**Comment by yam-liu** [comment]
有两点优化建议：
1. 在 Alfred 中输入 salad 时可不可以像 有道一样不需要额外点一下就自动执行操作。我看了有道的实现是用了 run behavior 设置的直接执行，这个第 2 个问题如果解了应该也可以做到？
2. 可不可以不用剪切板来传递参数，这样会污染剪切板历史

**Comment by crimx** [comment]
不用剪贴板传递需要安装本地脚本通讯，这个使用有一定门槛。我最近没有时间弄这个，有兴趣的可以看一下（[Firefox](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging), [Chrome](https://developer.chrome.com/extensions/nativeMessaging)）。

另外一个思路是可不可以让工具在查词后把剪贴板恢复。

**Comment by futuping** [comment]
这个不知道是不是bug，在我的catalina上，只有在谷歌浏览器窗口上才能成功显示salad的独立窗口，离开谷歌浏览器就不能正常显示，比如在mac预览app上显示出一个放大镜（如图）。已尝试：改了多次快捷键都不成功。
![image](https://user-images.githubusercontent.com/45912467/77247980-00475300-6c71-11ea-8b14-730797b07aeb.png)
麻烦shang大有空帮忙看看。谢谢！

**Comment by xinshangshangxin** [comment]
@futuping 
看下快捷键设置对不对, 最好类似截个图

https://user-images.githubusercontent.com/8779091/66553493-caf4f000-eb7d-11e9-84e8-933d6abf3985.png

**Comment by LucasZhan** [comment]
你好，我已经按照以上步骤设置完成。但在浏览器外依然无法使用salad划词翻译。触发command+control+l快捷键之后疑似会对选中内容重复循环进行copy操作而无法正常调出salad查词的触发图标（我观察到我另一个检测剪切板的软件被不断调用）。 同时左上角“edit”选项不断闪烁（这也是我感觉在进行copy操作的另一个原因），且尝试进入alfred软件电脑会变卡。 请问如何解决？

安装workflow之后貌似有个窗口弹出，但不知道是什么内容（不知道怎么的就没了）。会不会是权限申请类窗口？

环境：
Catalina 10.15.4
Firefox 74.0
salad查词 7.9.3
workflow 1.1
Alfred 4.0.9

**Reply by ChuYang-FE** [reply]
the same

**Reply by lasisal** [reply]
设置hotkey那里不能设置成control+command+L，因为关联的的AppleScript里面的内容是模拟按下control+command+L通过chrome呼起独立窗口的这个过程。这个时候再按这个快捷键组合就会进入一个死循环

**Comment by xinshangshangxin** [comment]
尝试手动触发 搜索:

1. 复制 一些文字(任意) 到剪切板中
2. 手动触发 快捷键 `command+control+l` 

如果此时 显示了 `saladict` 的话说明 `saladict` 和 chrome 快捷键部分配置没有问题

3. 关闭 你的另一个剪切板检测软件后 在尝试是否触发(可能两个软件打架了)

**Comment by LucasZhan** [comment]
你好。
123均已按顺序尝试
Com+ctrl+l无法调出saladict



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: SHANG殇 <notifications@github.com&gt;
发送时间: 2020年3月27日 20:42
收件人: crimx/ext-saladict <ext-saladict@noreply.github.com&gt;
抄送: LucasZhan <1012999614@qq.com&gt;, Comment <comment@noreply.github.com&gt;
主题: 回复：[crimx/ext-saladict] 技巧分享：Alfred Workflows 调用 Saladict 实现全局【文本翻译】 (#509)





 
尝试手动触发 搜索:
  
复制 一些文字(任意) 到剪切板中 

手动触发 快捷键 command+control+l 
 
如果此时 显示了 saladict 的话说明 saladict 和 chrome 快捷键部分配置没有问题
  
关闭 你的另一个剪切板检测软件后 在尝试是否触发(可能两个软件打架了)

**Comment by LucasZhan** [comment]
<img width="839" alt="Screen Shot 2020-03-27 at 8 55 00 pm" src="https://user-images.githubusercontent.com/41333680/77746707-4899b300-706d-11ea-84a7-3d1b519616cc.png">
<img width="740" alt="Screen Shot 2020-03-27 at 8 54 36 pm" src="https://user-images.githubusercontent.com/41333680/77746716-4cc5d080-706d-11ea-9fc3-82c84fc1bc37.png">

**Comment by LucasZhan** [comment]
> 你好。 123均已按顺序尝试 Com+ctrl+l无法调出saladict 发自我的iPhone
> […](#)
> ------------------ 原始邮件 ------------------ 发件人: SHANG殇 <notifications@github.com&gt; 发送时间: 2020年3月27日 20:42 收件人: crimx/ext-saladict <ext-saladict@noreply.github.com&gt; 抄送: LucasZhan <1012999614@qq.com&gt;, Comment <comment@noreply.github.com&gt; 主题: 回复：[crimx/ext-saladict] 技巧分享：Alfred Workflows 调用 Saladict 实现全局【文本翻译】 (#509) 尝试手动触发 搜索: 复制 一些文字(任意) 到剪切板中 手动触发 快捷键 command+control+l 如果此时 显示了 saladict 的话说明 saladict 和 chrome 快捷键部分配置没有问题 关闭 你的另一个剪切板检测软件后 在尝试是否触发(可能两个软件打架了) — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

尝试用chrome浏览器之后在浏览器外现已经可以通过快捷键调用saladict了 但是不能划词触发，必须要用快捷键

**Comment by xinshangshangxin** [comment]
> > 你好。 123均已按顺序尝试 Com+ctrl+l无法调出saladict 发自我的iPhone
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: SHANG殇 <[notifications@github.com](mailto:notifications@github.com)> 发送时间: 2020年3月27日 20:42 收件人: crimx/ext-saladict <[ext-saladict@noreply.github.com](mailto:ext-saladict@noreply.github.com)> 抄送: LucasZhan <[1012999614@qq.com](mailto:1012999614@qq.com)>, Comment <[comment@noreply.github.com](mailto:comment@noreply.github.com)> 主题: 回复：[crimx/ext-saladict] 技巧分享：Alfred Workflows 调用 Saladict 实现全局【文本翻译】 (#509) 尝试手动触发 搜索: 复制 一些文字(任意) 到剪切板中 手动触发 快捷键 command+control+l 如果此时 显示了 saladict 的话说明 saladict 和 chrome 快捷键部分配置没有问题 关闭 你的另一个剪切板检测软件后 在尝试是否触发(可能两个软件打架了) — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
> 
> 尝试用chrome浏览器之后在浏览器外现已经可以通过快捷键调用saladict了 但是不能划词触发，必须要用快捷键

"划词触发" 本身就不支持浏览器外
`Alfred Workflows` 支持的是 快捷键查询

**Comment by LucasZhan** [comment]
> > > 你好。 123均已按顺序尝试 Com+ctrl+l无法调出saladict 发自我的iPhone
> > > […](#)
> > > ------------------ 原始邮件 ------------------ 发件人: SHANG殇 <[notifications@github.com](mailto:notifications@github.com)> 发送时间: 2020年3月27日 20:42 收件人: crimx/ext-saladict <[ext-saladict@noreply.github.com](mailto:ext-saladict@noreply.github.com)> 抄送: LucasZhan <[1012999614@qq.com](mailto:1012999614@qq.com)>, Comment <[comment@noreply.github.com](mailto:comment@noreply.github.com)> 主题: 回复：[crimx/ext-saladict] 技巧分享：Alfred Workflows 调用 Saladict 实现全局【文本翻译】 (#509) 尝试手动触发 搜索: 复制 一些文字(任意) 到剪切板中 手动触发 快捷键 command+control+l 如果此时 显示了 saladict 的话说明 saladict 和 chrome 快捷键部分配置没有问题 关闭 你的另一个剪切板检测软件后 在尝试是否触发(可能两个软件打架了) — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.
> > 
> > 
> > 尝试用chrome浏览器之后在浏览器外现已经可以通过快捷键调用saladict了 但是不能划词触发，必须要用快捷键
> 
> "划词触发" 本身就不支持浏览器外
> `Alfred Workflows` 支持的是 快捷键查询

谢谢， 如果使用firefox的话该如何在浏览器外使用这个workflow呢

**Comment by xinshangshangxin** [comment]
https://saladict.crimx.com/manual.html#shortcuts

Firefox 66 开始支持，在扩展管理页面 about:addons 上方的齿轮设置按钮点开最底 Manage Extension Shortcuts。

同理设置 全局快捷键

由于我不使用 Firefox, 未测试

**Comment by LucasZhan** [comment]
> https://saladict.crimx.com/manual.html#shortcuts
> 
> Firefox 66 开始支持，在扩展管理页面 about:addons 上方的齿轮设置按钮点开最底 Manage Extension Shortcuts。
> 
> 同理设置 全局快捷键
> 
> 由于我不使用 Firefox, 未测试

谢谢， firefox在扩展管理页面 about:addons无法设置全局快捷键。在这里提醒一下后来人，此workflow建议用chrome。

**Comment by drcege** [comment]
说实话总是按快捷键不太方面，而且我在 mac 上配置了全局快捷键不知为何也没生效。

解决浏览器外查词最好还是将沙拉查词包装成一个单独的 app 吧，可以直接在系统层面使用。

**Comment by xiaoming-w** [comment]
老问题，按照步骤一步步操作，但是两者都没有反应： Alfred快捷键和alfred的salad搜索，希望给予建议，谢谢
![image](https://user-images.githubusercontent.com/43328230/84852958-20966c00-b02c-11ea-9da4-4f0446a14939.png)

![image](https://user-images.githubusercontent.com/43328230/84852980-2c822e00-b02c-11ea-8d53-738695a93db5.png)
![image](https://user-images.githubusercontent.com/43328230/84853002-31df7880-b02c-11ea-969d-1a2703781a7a.png)
![image](https://user-images.githubusercontent.com/43328230/84853021-3b68e080-b02c-11ea-810e-a85ba76996a9.png)
![image](https://user-images.githubusercontent.com/43328230/84853064-56d3eb80-b02c-11ea-97c1-b61881a53807.png)

**Reply by jianJizz** [reply]
我之前也是这个问题，是mac的权限问题，现在可以了
![image](https://user-images.githubusercontent.com/24221653/114170746-efde4500-9965-11eb-9298-2bb110bfff97.png)

**Comment by najunuoyan** [comment]
这个脚本真NB，配合一个可以自定义按钮功能的鼠标，直接全局鼠标一键查词，方便到不行，感谢

**Comment by Hackerer** [comment]
报这个错，执行不下去了，怎么解决

[15:17:31.994] 沙拉[Hotkey] Processing complete
[15:17:32.003] 沙拉[Hotkey] Passing output '' to Run NSAppleScript
[15:17:32.230] ERROR: 沙拉[Run NSAppleScript] {
    NSAppleScriptErrorAppName = "System Events";
    NSAppleScriptErrorBriefMessage = "\U201cAlfred\U201d\U4e0d\U5141\U8bb8\U53d1\U9001\U6309\U952e\U3002";
    NSAppleScriptErrorMessage = "\U201cSystem Events\U201d\U9047\U5230\U4e00\U4e2a\U9519\U8bef\Uff1a\U201cAlfred\U201d\U4e0d\U5141\U8bb8\U53d1\U9001\U6309\U952e\U3002";
    NSAppleScriptErrorNumber = 1002;
    NSAppleScriptErrorRange = "NSRange: {69, 44}";
}

![image](https://user-images.githubusercontent.com/15344671/86735366-55af4080-c065-11ea-8c7a-ee03b22a4329.png)

**Comment by xinshangshangxin** [comment]
> \U201cAlfred\U201d\U4e0d\U5141\U8bb8\U53d1\U9001\U6309\U952e\U3002
> “Alfred”不允许发送按键。

看看 alfred 的 权限对不对

![image](https://user-images.githubusercontent.com/8779091/86736081-d4a47900-c065-11ea-8008-5edfc638c7b0.png)

**Comment by Hackerer** [comment]
![image](https://user-images.githubusercontent.com/15344671/86736920-7a57e800-c066-11ea-8c38-87885be49307.png)
权限应该是OK的

**Comment by xinshangshangxin** [comment]
尝试下  先去掉勾选, 然后再重新候选下呢?

**Comment by xinshangshangxin** [comment]
第一次使用应该有这个提示
![image](https://user-images.githubusercontent.com/8779091/86737921-3a453500-c067-11ea-822c-873375618ff4.png)

**Comment by Hackerer** [comment]
好了，多谢，应该是我之前重装了一次Alfred，导致隐私设置里没有更新，我把这个权限去掉后，重新加上就好了

**Comment by A-stupid-fish** [comment]
设置完成后，快捷键触发弹出独立窗口没有问题，但是所查单词是上一次复制到剪贴板上的，而不是触发前选中的单词，请教各位大神该如何解决？

**Reply by FakeKuryr** [reply]
the same,  你解决了吗

**Comment by okerivy** [comment]
> 设置完成后，快捷键触发弹出独立窗口没有问题，但是所查单词是上一次复制到剪贴板上的，而不是触发前选中的单词，请教各位大神该如何解决？

1, 这里面是有两个快捷键的，一个是沙拉查词 插件 的快捷健  ctrl + cmd + L
另一个是alfred 的快捷键 ，这个不能设置成上面那个，因为 alfred 是调用 那个快捷键的。需要设置成其他的，例如 double cmd

2, 沙拉查词 插件 的快捷健  ctrl + cmd + L  不会自动复制当前选中 的文字，所以弹窗查出来的是上次的词。
这也是为啥用 alfred的原因，他是选中后，按下刚才的快捷键  double cmd  就会 自动复制，并弹出窗口。

3，当然2的问题，可能是沙拉这个插件内部，还有其他设置项，但是没研究。

**Reply by huang-zi-jian** [reply]
我设置的全局快捷键shift+cmd+L，alfred快捷键shift+cmd+D，选中后调用alfred会自动复制，但是没有弹出窗口这是什么原因呢，Alfred的权限是ok的

**Reply by AndyFred0930** [reply]
同上面的问题

**Comment by Mamba4ever** [comment]
请问大神设置了快捷键却无法选择全局怎么解决啊？原来用WIN直接就可以了 现在是macOS,难道是系统的问题吗

![Uploading 截屏2020-12-02 下午8.02.47.png…]()

**Comment by PerfectDay20** [comment]
Firefox没有全局快捷键, 一个方法是在脚本中先激活Firefox, 然后脚本中的快捷键就会对Firefox有效了, 比如:
```
on alfred_script(q)
  tell application "Firefox" to activate

  tell application "System Events"
	# 快捷键打开沙拉词典
	key code 37 using {control down, command down}
  end tell
end alfred_script
```

**Comment by ShawnWx2019** [comment]
**哇，应该您这个是原创把！**
我是看了某公众号的文章跟着做的`alfred+Saladict`, 我看了`alfred workflows`里面的脚本都和您这儿差不多，只不过把快捷键改成的`cmd+shift+d`，新版edge可以用google的插件，所以我在edge下搞的，edge下需要打开插件访问剪切板的权限，同时把快捷键改成全局，然后载multitouch下面选择一个trackpad的手势或者鼠标手势，录制`cmd+shift+d`,选中文本后就可以通过手势来直接启动saladict。
**非常棒的教程！**

**Comment by gabrielle2324** [comment]
解压上面下载的 saladict.alfredworkflow.zip, 后无法双击 saladict.alfredworkflow ， 说文件的格式比正确，破损了，也看不到alfred workflow，我该怎么办尼

**Comment by jeremy-ww** [comment]
这个版本可以在 Chrome 没有启动、隐藏的情况下直接唤起独立弹窗。


[沙拉.zip](https://github.com/crimx/ext-saladict/files/6162386/default.zip)

**Comment by xianchenxc** [comment]
有 alfred 3 的 workflow 吗？

**Comment by FlameSanchez** [comment]
妈的，Alfred 4哪来的也不说一声

**Comment by huang-zi-jian** [comment]
我设置的全局快捷键shift+cmd+L，alfred快捷键shift+cmd+D，并且选中后cmd+C复制之后再shift+cmd+L是可以有弹窗的。但是选中后直接调用alfred快捷键并没有弹出窗口，只能自动复制，这是什么原因呢，Alfred的权限是ok的，

**Comment by HXDreamChaser** [comment]
Mac mini app store 里面装的Alfred上没有workflows功能按钮！

请问如何让Alfred上出现workflows按钮？我想在mac上的其他pdf阅读软件上使用沙拉划词翻译。
另外，我使用Alfred打开saladict.alfredworkflow并没有反应。


## 截图
这是我的Alfred截图，比别人的在最上面一行少了workflows功能按钮
![Alfred](https://user-images.githubusercontent.com/29406836/192134973-095e9ea6-e195-47de-8838-8541ead86abe.png)

这是网上看到的别人的Alfred截图：
![image](https://user-images.githubusercontent.com/29406836/192135074-e9ebe180-360e-40e2-95c3-aef10a7568fd.png)

**Reply by Somiona** [reply]
在alfred官网下载安装，并且付费购买alfred power pack，这个workflow是一个需要单独付费购买的功能

**Comment by zhjrcc** [comment]
我想知道，能不能在进行翻译前对翻译内容先做一点修剪的？原因是很多英文的PDF，每一行单词结尾如果单词不完整，则会有一个连字符"- "，有时候直接翻译会导致翻译出错的问题，如果能在翻译前将这种"- "替换成""就好了。

**Comment by m1a0cX** [comment]
请问为什么在chrome外唤醒沙拉查词时出现的独立窗口会自动跳转到第一个桌面，而不是在当前应用窗口下显示？有大佬知道这应该怎么解决吗？
